src/main/java/com/example/bank/controller/TransactionController.java

```diff
@@ -1,43 +1,66 @@
 package com.example.bank.controller;
 
-import com.example.bank.entity.Transaction;
-import com.example.bank.repository.TransactionRepository;
+import com.example.bank.DTO.AccountDTO;
+import com.example.bank.entity.Account;
+import com.example.bank.service.Impl.AccountServiceImpl;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.ResponseBody;
-
-import java.lang.reflect.Array;
-import java.util.ArrayList;
-import java.util.HashMap;
+import org.springframework.web.bind.annotation.*;
+
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
-import java.util.stream.StreamSupport;
-
-@Controller
 
+@RestController
+@RequestMapping("/transaction")
```

Comment:
use "/transactions" here (see use constructor injection instead of field injection)